ibis/backends/mssql/tests/test_client.py

```diff
@@ -241,3 +241,27 @@ def test_from_url():
     )
     result = new_con.sql("SELECT 1 AS [a]").to_pandas().a.iat[0]
     assert result == 1
+
+
+def test_dot_sql_with_unnamed_columns(con):
```

Comment:
Gonna see about making this backend-agnostic to catch other cases of this fun and annoying backend behavior.